### PR TITLE
TQDM progressbar for genai metrics

### PR DESCRIPTION
**Pull request by prithvikannan**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/prithvikannan/mlflow/pull/10203?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10203/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10203
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

```
2023/10/27 16:43:06 INFO mlflow.models.evaluation.default_evaluator: Evaluating metrics: relevance
  0%|                                                                                                    | 0/4 [00:00<?, ?it/s]2023/10/27 16:43:06 WARNING mlflow.openai.api_request_parallel_processor: Encountered rate limit error. Pausing to cool down for 14.999726057052612 seconds...
2023/10/27 16:43:06 WARNING mlflow.openai.api_request_parallel_processor: Encountered rate limit error. Pausing to cool down for 14.99975323677063 seconds...
2023/10/27 16:43:06 WARNING mlflow.openai.api_request_parallel_processor: Encountered rate limit error. Pausing to cool down for 14.999823808670044 seconds...
 25%|███████████████████████                                                                     | 1/4 [00:08<00:25,  8.35s/it]2023/10/27 16:43:21 WARNING mlflow.openai.api_request_parallel_processor: Encountered rate limit error. Pausing to cool down for 0.0648488998413086 seconds...
2023/10/27 16:43:21 WARNING mlflow.openai.api_request_parallel_processor: Encountered rate limit error. Pausing to cool down for 0.07401394844055176 seconds...
2023/10/27 16:43:21 WARNING mlflow.openai.api_request_parallel_processor: Encountered rate limit error. Pausing to cool down for 0.05842185020446777 seconds...
2023/10/27 16:43:21 WARNING mlflow.openai.api_request_parallel_processor: Encountered rate limit error. Pausing to cool down for 14.999640703201294 seconds...
2023/10/27 16:43:21 WARNING mlflow.openai.api_request_parallel_processor: Encountered rate limit error. Pausing to cool down for 14.999744176864624 seconds...
2023/10/27 16:43:29 WARNING mlflow.openai.api_request_parallel_processor: 2 rate limit errors received. Consider running at a lower rate.
 50%|██████████████████████████████████████████████                                              | 2/4 [00:23<00:25, 12.51s/it]2023/10/27 16:43:36 WARNING mlflow.openai.api_request_parallel_processor: Encountered rate limit error. Pausing to cool down for 0.07663202285766602 seconds...
2023/10/27 16:43:36 WARNING mlflow.openai.api_request_parallel_processor: Encountered rate limit error. Pausing to cool down for 0.0563962459564209 seconds...
2023/10/27 16:43:42 WARNING mlflow.openai.api_request_parallel_processor: 4 rate limit errors received. Consider running at a lower rate.
 75%|█████████████████████████████████████████████████████████████████████                       | 3/4 [00:36<00:12, 12.75s/it]2023/10/27 16:43:43 WARNING mlflow.openai.api_request_parallel_processor: 4 rate limit errors received. Consider running at a lower rate.
100%|████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:36<00:00,  9.23s/it]
{'toxicity/v1/mean': 0.00018985224596690387, 'toxicity/v1/variance': 2.445330018924171e-09, 'toxicity/v1/p90': 0.0002462980875861831, 'toxicity/v1/ratio': 0.0, 'flesch_kincaid_grade_level/v1/mean': 8.0, 'flesch_kincaid_grade_level/v1/variance': 21.705000000000002, 'flesch_kincaid_grade_level/v1/p90': 12.910000000000002, 'ari_grade_level/v1/mean': 9.55, 'ari_grade_level/v1/variance': 29.762499999999992, 'ari_grade_level/v1/p90': 15.540000000000001, 'faithfulness/v1/mean': 3.0, 'faithfulness/v1/variance': 4.0, 'faithfulness/v1/p90': 5.0, 'relevance/v1/mean': 4.25, 'relevance/v1/variance': 0.1875, 'relevance/v1/p90': 4.7}
Downloading artifacts: 100%|███████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 1527.42it/s]
```

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

Manual

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @prithvikannan !

**Review by sunishsheth2009 — APPROVED** [review]

**Comment by harupy** [comment]
This change can lead to ImportError. Did we add tqdm in mlflow dependecnies in other PR?

File: mlflow/metrics/genai/genai_metric.py [hunk 1]
```diff
@@ -328,11 +328,20 @@ def score_model_on_one_payload(
...
+                    indx = futures[future]
+                    score, justification = future.result()
+                    scores[indx] = score
+                    justifications[indx] = justification
+            except ImportError:
+                for future in as_completed(futures):
+                    indx = futures[future]
+                    score, justification = future.result()
+                    scores[indx] = score
+                    justifications[indx] = justification
```
Comment (harupy) [inline mlflow/metrics/genai/genai_metric.py:344]:
```suggestion
            as_comp = as_completed(futures)
            try:
                from tqdm import tqdm

                as_comp = tqdm(as_comp)
            except ImportError:
                pass

            for future in as_comp:
                indx = futures[future]
                score, justification = future.result()
                scores[indx] = score
                justifications[indx] = justification
```

**Review by harupy — APPROVED** [review]
LGTM!

File: mlflow/metrics/genai/genai_metric.py [hunk 2]
```diff
@@ -328,7 +328,15 @@ def score_model_on_one_payload(
                 for indx, (input, output) in enumerate(zip(inputs, outputs))
             }
 
-            for future in as_completed(futures):
+            as_comp = as_completed(futures)
+            try:
+                from tqdm.auto import tqdm
+
+                as_comp = tqdm(as_comp)
```
Comment (harupy) [inline mlflow/metrics/genai/genai_metric.py:335]:
```suggestion
                as_comp = tqdm(as_comp, total=len(futures))
```

**Pull request merged**